File: src/types/project.ts
export interface Project {
  id: string;
  name: string;
  path: string;
  label?: string;
  branches: Branch[];
  remote_branches?: string[];
  worktrees: Worktree[];
}

export interface Branch {
  id: string;
  name: string;
  label?: string;
  project_id: string;
  terminals: Terminal[];
}

export interface Worktree {
  id: string;
  branch: string;
  path: string;
  label?: string;
  project_id: string;
  status: "Clean" | "Dirty" | "Detached";
  terminals: Terminal[];
}

export interface Terminal {
  id: string;
  name: string;
  label?: string;
  parent_id: string;
  parent_type: "branch" | "worktree";
  project_id: string;
  pty_id?: string;
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};
use std::path::PathBuf;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Project {
    pub id: String,
    pub name: String,
    pub path: PathBuf,
    pub label: Option<String>,
    pub branches: Vec<Branch>,
    #[serde(default)]
    pub remote_branches: Vec<String>,
    pub worktrees: Vec<Worktree>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Branch {
    pub id: String,
    pub name: String,
    pub label: Option<String>,
    pub project_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Worktree {
    pub id: String,
    pub branch: String,
    pub path: PathBuf,
    pub label: Option<String>,
    pub project_id: String,
    pub status: WorktreeStatus,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum WorktreeStatus {
    Clean,
    Dirty,
    Detached,
}
